src/pykeen/nn/combinations.py

```diff
@@ -25,6 +26,79 @@
     "GatedCombination",
 ]
 
+logger = logging.getLogger(__name__)
+
+
+class NCombination(nn.Module, ABC):
+    """Base class for combinations."""
+
+    @abstractmethod
+    def forward(self, xs: Sequence[torch.FloatTensor]) -> torch.FloatTensor:
+        """
+        Combine a sequence of individual representations.
+
+        :param xs: shape: `(*batch_dims, *input_dims_i)`
+            the individual representations
+
+        :return: shape: `(*batch_dims, *output_dims)`
+            a combined representation
+        """
+        raise NotImplementedError
+
+    def output_shape(self, input_shapes: Sequence[Tuple[int, ...]]) -> Tuple[int, ...]:
+        """
+        Calculate the outpu shape for the given input shapes.
+
+        .. note ::
+            this method runs a single forward pass if no symbolic computation is available.
+
+        :param input_shapes:
+            the input shapes without the batch dimensions
+
+        :return:
+            the output shape
+        """
+        logger.warning("No symbolic computation of output shape.")
+        return self(xs=[torch.empty(size=shape) for shape in input_shapes]).shape
```

Comment:
isn't there like a symbolic "meta"-device now for this kind of operation?

Reply:
> NotImplementedError: Could not run 'aten::_cat' with arguments from the 'Meta' backend. This could be because the operator doesn't exist for this backend

As soon as you start calculating with it, it will break.

Reply:
😿

Reply:
but this implementation is fine as it is too